tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+    );
+    /** Regex to match the binary name of a class (e.g. "package.submodule.InnerClass$OuterClass") */
+    private static final Pattern PACKAGE_CLASS = Pattern.compile("[a-zA-Z_][a-zA-Z\\d_]*(\\.[a-zA-Z_][a-zA-Z\\d_]*)*");
+
+    /** Private constructor to avoid creating an instance of this class. */
+    private JavaParserUtils() {
+        throw new UnsupportedOperationException("This class cannot be instantiated.");
+    }
+
+    public static JavaParser getJavaParser() {
+        if (javaParser == null) {
+            String root = "src/main/resources/projects-packaged";
+            SymbolSolverCollectionStrategy strategy = new SymbolSolverCollectionStrategy();
+            strategy.collect(Paths.get(root));
+            javaParser = new JavaParser();
+            javaParser.getParserConfiguration().setSymbolResolver(strategy.getParserConfiguration().getSymbolResolver().get());
+        }
+        return javaParser;
+    }
+
+    /**
+     * Gets a synthetic empty JavaParser class.
+     *
+     * @return an empty JavaParser class
+     */
+    private static TypeDeclaration<?> getSyntheticClass() {
+        return javaParser.parse(SYNTHETIC_CLASS_SOURCE).getResult().orElseThrow()
+                .getLocalDeclarationFromClassname(SYNTHETIC_CLASS_NAME).get(0);
+    }
+
+    /**
+     * Adds a synthetic empty method block to a given class.
+     *
+     * @param typeDeclaration the class to add a method to
+     * @return an empty JavaParser method block in the given class
+     */
+    private static BlockStmt getSyntheticBlockStmt(TypeDeclaration<?> typeDeclaration) {
+        return typeDeclaration.addMethod(SYNTHETIC_METHOD_NAME).getBody().orElseThrow();
+    }
+
+    /**
+     * Creates a "java.lang.Object" type.
+     *
+     * @return a "java.lang.Object" type
+     */
+    public static ResolvedType getObjectType() {
+        TypeDeclaration<?> syntheticClass = getSyntheticClass();
+        BlockStmt syntheticMethod = getSyntheticBlockStmt(syntheticClass);
+        return syntheticMethod
+                .addStatement("java.lang.Object objectVar;")
+                .getStatements().getLast().orElseThrow()
+                .asExpressionStmt().getExpression()
+                .asVariableDeclarationExpr().getVariables().get(0)
+                .resolve().getType();
+    }
+
+    /**
+     * Injects a new synthetic method declaration into a class given an
+     * original method under test, including:
+     * <ul>
+     *     <li>variable declaration for each method argument in the original method</li>
+     *     <li>variable declaration for the return type of the original method</li>
+     *     <li>variable declaration (of unknown type) using given expression</li>
+     * </ul>
+     *
+     * @param jpClass the declaring class in which to insert a synthetic method
+     * @param jpCallable the method under analysis
+     * @param methodArgs tokens for the parameters of {@code jpCallable}
```

Comment:
Functionality for this is implemented in DatasetUtils.java in the next pull request, as `methodArgs` is a feature in the dataset. Should we re-retrieve this information in the dataset for the implementation in the second pull request? I can make note of it here and implement that in the next pull request. However, as DatasetUtils.java does not yet exist in this pull request, it would be difficult to do here.